### PR TITLE
zh-translation: docs/root/faq/debugging/why_is_envoy_sending_http2_re…

### PR DESCRIPTION
**Pull request by vgbhfive**
…sets.rst

For an explanation of how to fill out the fields, please see the relevant section
in [PULL_REQUESTS.md](https://github.com/envoyproxy/envoy/blob/master/PULL_REQUESTS.md)

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes: docs/root/faq/debugging/why_is_envoy_sending_http2_resets.rst
Release Notes:
[Optional Runtime guard:]
[Optional Fixes #Issue] #710 
[Optional Deprecated:]

### CODE REVIEW

File: docs/root/faq/debugging/why_is_envoy_sending_http2_resets.rst [hunk 1]
```diff
@@ -1,24 +1,17 @@
...
-Why is Envoy sending HTTP/2 resets?
+为什么 Envoy 要发送 HTTP/2 重置？
 ===================================
 
-The HTTP/2 reset path is mostly governed by the codec Envoy uses to frame HTTP/2, nghttp2. nghttp2 has
-extremely good adherence to the HTTP/2 spec, but as many clients are not exactly as compliant, this
-mismatch can cause unexpected resets. Unfortunately, unlike the debugging the 
-:ref:`internal response path <why_is_envoy_sending_internal_responses>`, Envoy has limited visibility into
-the specific reason nghttp2 reset a given stream.
+HTTP/2 重置路径主要由 Envoy 编解码器使用 HTTP/2、nghttp2 来控制。
```
Comment (helight) [inline docs/root/faq/debugging/why_is_envoy_sending_http2_resets.rst]:
```suggestion
HTTP/2 重置路径主要是由 Envoy 封装 HTTP/2、nghttp2 的编解码器来控制。
```

File: docs/root/faq/debugging/why_is_envoy_sending_http2_resets.rst [hunk 2]
```diff
@@ -1,24 +1,17 @@
...
+nghttp2 对于 HTTP/2 有着非常好的遵从性，但是由于许多客户端并不完全遵从，因此这种不完全匹配就会导致以外的重置。
+不幸的是，与调试 :ref:`内部响应路径 <why_is_envoy_sending_internal_responses>` 不同，Envoy 在给定具体原因的 nghttp2 中重置流有着可限的能见度。
```
Comment (helight) [inline docs/root/faq/debugging/why_is_envoy_sending_http2_resets.rst]:
```suggestion
不幸的是，与调试 :ref:`内部响应路径 <why_is_envoy_sending_internal_responses>` 不同，Envoy 在 nghttp2 重置给定流的具体原因可见度有限。
```

File: docs/root/faq/debugging/why_is_envoy_sending_http2_resets.rst [hunk 3]
```diff
@@ -1,24 +1,17 @@
...
 
-If you have a reproducible failure case, you can run it against a debug Envoy with "-l trace" to get
-detailed nghttp2 error logs, which often indicate which header failed compliance checks. Alternately,
-if you can afford to run with "-l trace" on a machine encountering the errors, you can look for logs
-from the file "source/common/http/http2/codec_impl.cc" of the form
-`invalid http2: [nghttp2 error detail]`
-for example:
-`invalid http2: Invalid HTTP header field was received: frame type: 1, stream: 1, name: [content-length], value: [3]`
+如果你有一个可再现的失败案例，你可以在带有 "-l trace" 的 Envoy 中调试运行它，那么你将获得详细的 nghttp2 异常日志，这些日志通常会指出哪些头信息未通过符合性检查。
```
Comment (helight) [inline docs/root/faq/debugging/why_is_envoy_sending_http2_resets.rst]:
```suggestion
如果你有一个可再现的失败案例，你可以用 "-l trace" 参数，在 debug 版本的 Envoy 中运行它，以便将获得详细的 nghttp2 错误日志，这些日志通常会指出哪些头信息未通过符合性检查。
```

File: docs/root/faq/debugging/why_is_envoy_sending_http2_resets.rst [hunk 4]
```diff
@@ -1,24 +1,17 @@
...
+或者，如果你也可以在一台遇到异常的机器上带有 "-l trace" 运行，那么你可以从格式为 `invalid http2: [nghttp2 error detail]` 的文件 "source/common/http/http2/codec_impl.cc" 中找到日志。
```
Comment (helight) [inline docs/root/faq/debugging/why_is_envoy_sending_http2_resets.rst]:
```suggestion
或者，如果你可以在一台遇到异常的机器上用 "-l trace" 参数运行，那么你可以从文件 "source/common/http/http2/codec_impl.cc" 中寻找格式为 `invalid http2: [nghttp2 error detail]` 的日志。
```

File: docs/root/faq/debugging/why_is_envoy_sending_http2_resets.rst [hunk 5]
```diff
@@ -1,24 +1,17 @@
...
+例如：`invalid http2: Invalid HTTP header field was received: frame type: 1, stream: 1, name: [content-length], value: [3]`
 
-You can also check :ref:`HTTP/2 stats`<config_http_conn_man_stats_per_codec>`: in many cases where
-Envoy resets streams, for example if there are more headers than allowed by configuration or flood
-detection kicks in, http2 counters will be incremented as the streams are reset.
+另外你也可以检查 :ref:`HTTP/2 状态`<config_http_conn_man_stats_per_codec>`: 在许多情况下，Envoy 会重置流。
+例如如果有更多的头信息超过配置的允许界限或流检查点启用，那么 http2 计数器将会增加，这时流也会被重置。
```
Comment (helight) [inline docs/root/faq/debugging/why_is_envoy_sending_http2_resets.rst]:
```suggestion
例如如果有更多的头信息超过配置允许的界限或启用了流检查点，那么 http2 计数器将会增加，这时流也会被重置。
```

**Review by helight — COMMENTED** [review]
一点建议

File: docs/root/faq/debugging/why_is_envoy_sending_http2_resets.rst (hunk 5 above)
Comment (vgbhfive) [reply docs/root/faq/debugging/why_is_envoy_sending_http2_resets.rst]: 感谢指正。

File: docs/root/faq/debugging/why_is_envoy_sending_http2_resets.rst [hunk 6]
```diff
@@ -1,24 +1,17 @@
...
+为什么 Envoy 要发送 HTTP/2 重置？
 ===================================
 
-The HTTP/2 reset path is mostly governed by the codec Envoy uses to frame HTTP/2, nghttp2. nghttp2 has
-extremely good adherence to the HTTP/2 spec, but as many clients are not exactly as compliant, this
-mismatch can cause unexpected resets. Unfortunately, unlike the debugging the 
-:ref:`internal response path <why_is_envoy_sending_internal_responses>`, Envoy has limited visibility into
-the specific reason nghttp2 reset a given stream.
+HTTP/2 重置路径主要是由 Envoy 封装 HTTP/2、nghttp2 的编解码器来控制。
+nghttp2 对于 HTTP/2 有着非常好的遵从性，但是由于许多客户端并不完全遵从，因此这种不完全匹配就会导致以外的重置。
```
Comment (addozhang) [inline docs/root/faq/debugging/why_is_envoy_sending_http2_resets.rst]:
```suggestion
nghttp2 严格遵守 HTTP/2 规范，但是由于许多客户端不完全兼容，这种不匹配会导致意外重置。
```

File: docs/root/faq/debugging/why_is_envoy_sending_http2_resets.rst [hunk 7]
```diff
@@ -1,24 +1,17 @@
...
+不幸的是，与调试 :ref:`内部响应路径 <why_is_envoy_sending_internal_responses>` 不同，Envoy 在 nghttp2 中重置流的具体原因可见度有限。
 
-If you have a reproducible failure case, you can run it against a debug Envoy with "-l trace" to get
-detailed nghttp2 error logs, which often indicate which header failed compliance checks. Alternately,
-if you can afford to run with "-l trace" on a machine encountering the errors, you can look for logs
-from the file "source/common/http/http2/codec_impl.cc" of the form
-`invalid http2: [nghttp2 error detail]`
-for example:
-`invalid http2: Invalid HTTP header field was received: frame type: 1, stream: 1, name: [content-length], value: [3]`
+如果你有一个可再现的失败案例，你可以用 "-l trace" 参数，在 debug 版本的 Envoy 中运行它，以便将获得详细的 nghttp2 异常日志，这些日志通常会指出哪些头信息未通过符合性检查。
```
Comment (addozhang) [inline docs/root/faq/debugging/why_is_envoy_sending_http2_resets.rst]:
```suggestion
如果你有一个可再现的失败案例，你可以用 "-l trace" 参数，在 debug 版本的 Envoy 中运行它，以便将获得详细的 nghttp2 错误日志，这些日志通常会指出哪些头信息未通过合规检查。
```

File: docs/root/faq/debugging/why_is_envoy_sending_http2_resets.rst [hunk 8]
```diff
@@ -1,24 +1,17 @@
...
+或者，如果你可以在一台遇到异常的机器上用 "-l trace" 参数运行，那么你可以从文件 "source/common/http/http2/codec_impl.cc" 中寻找格式为 `invalid http2: [nghttp2 error detail]` 的日志。
+例如：`invalid http2: Invalid HTTP header field was received: frame type: 1, stream: 1, name: [content-length], value: [3]`
 
-You can also check :ref:`HTTP/2 stats`<config_http_conn_man_stats_per_codec>`: in many cases where
-Envoy resets streams, for example if there are more headers than allowed by configuration or flood
-detection kicks in, http2 counters will be incremented as the streams are reset.
+另外你也可以检查 :ref:`HTTP/2 状态`<config_http_conn_man_stats_per_codec>`: 在许多情况下，Envoy 会重置流。
```
Comment (addozhang) [inline docs/root/faq/debugging/why_is_envoy_sending_http2_resets.rst]:
```suggestion
另外你也可以检查 :ref:`HTTP/2 统计数据`<config_http_conn_man_stats_per_codec>`: 在许多情况下，Envoy 会重置流。
```

File: docs/root/faq/debugging/why_is_envoy_sending_http2_resets.rst [hunk 9]
```diff
@@ -1,24 +1,17 @@
...
+例如如果有更多的头信息超过配置允许的界限或启用了流检查点，那么 http2 计数器将会增加，这时流也会被重置。
```
Comment (addozhang) [inline docs/root/faq/debugging/why_is_envoy_sending_http2_resets.rst]:
```suggestion
例如，如果标头超过了配置所允许的范围，或者洪泛检测启动，则在重置流时，http2 计数器将增加。
```

**Review by addozhang — APPROVED** [review]
/LGTM

**Pull request merged**